### ISSUE TITLE
Error when appending to array that contained a tuple

### ISSUE THREAD

**Issue by zakissimo** [microsoft/pyright#3313]
**Describe the bug**
Pyright complains when trying to append to an array that previously contained a tuple.

**To Reproduce**
```python
foo = [("bar", "baz")]
_ = foo.pop()
foo.append("bar") # <-- Error (see below)
```
```sh
Pyright: Argument of type "Literal['bar']" cannot be assigned to parameter "__object" of type "tuple[Literal['bar'], Literal['baz']]" in function "append"
  "Literal['bar']" is incompatible with "tuple[Literal['bar'], Literal['baz']]"
```

**Expected behavior**
No error should be displayed

**VS Code extension or command-line**
![image](https://user-images.githubusercontent.com/64891486/162339156-a51b77ec-2644-42b1-af5b-f1d07dcbd9b9.png)

**Comment by JelleZijlstra** [comment]
Pyright is correct. You made a list of tuples, and you are putting a string in the list. If you want Pyright to infer a different type, you can explicitly declare it, e.g.:

```python
foo: list[object] = [("bar", "baz")]
_ = foo.pop()
foo.append("bar") # <-- Error (see below)
```

**Comment by zakissimo** [comment]
I understand, thanks for the clarification.
